src/filesystem.cc

```diff
@@ -1291,10 +1313,13 @@ namespace s3 = Aws::S3;
 
 class S3FileSystem : public FileSystem {
  public:
-  S3FileSystem(const Aws::SDKOptions& options, const std::string& s3_path);
+  S3FileSystem(
+      const std::string& s3_path, const std::string& secret_key,
+      const std::string& key_id, const std::string& region,
+      const std::string& session_token, const std::string& profile_name);
```

Comment:
Maybe encapsulate in a simple `S3Options` struct?

Reply:
Yes. Updated.